.github/workflows/release.yml

```
@@ -12,12 +12,10 @@ permissions:
   contents: read
 
 jobs:
-  release-pypi:
-    name: Upload release to PyPI
+  build:
+    name: Build release assets
     runs-on: ubuntu-latest
-    environment:
-      name: PyPI
-      url: https://pypi.org/project/pylint/
+    if: github.event_name == 'release' && startsWith(github.ref, 'refs/tags')
```

Is the combined check needed?
Isn't release always connected to a tag?
```suggestion
    if: github.event_name == 'release'
```

Also, there's now a convenient value for the ref type in the context:
```suggestion
    if: github.ref_type == 'tag'
```

---

Nice! Didn't know about `github.ref_type`.

> Isn't release always connected to a tag?

Yes, it should be. Created #10267 to just use `github.event_name == 'release'` for the check.